core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1714,16 +1741,20 @@ class WorkspaceService(protected val userInfo: UserInfo,
         userComment = submissionRequest.userComment
       )
 
-      // implicitly passed to SubmissionComponent.create
-      implicit val subStatusCounter = submissionStatusCounter(workspaceMetricBuilder(workspaceContext.toWorkspaceName))
-      implicit val wfStatusCounter = (status: WorkflowStatus) =>
-        if (config.trackDetailedSubmissionMetrics) Option(workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceContext.toWorkspaceName, submissionId))(status))
-        else None
-
-      dataAccess.submissionQuery.create(workspaceContext, submission)
+      logAndCreateDbSubmission(workspaceContext, submissionId, submission, dataAccess)
     }
   }
 
+  def logAndCreateDbSubmission(workspaceContext: Workspace, submissionId: UUID, submission: Submission, dataAccess: DataAccess): ReadWriteAction[Submission] = {
+    // implicitly passed to SubmissionComponent.create
+    implicit val subStatusCounter = submissionStatusCounter(workspaceMetricBuilder(workspaceContext.toWorkspaceName))
+    implicit val wfStatusCounter = (status: WorkflowStatus) =>
+      if (config.trackDetailedSubmissionMetrics) Option(workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceContext.toWorkspaceName, submissionId))(status))
+      else None
+
+    dataAccess.submissionQuery.create(workspaceContext, submission)
```

Comment:
I'm not sure if you're asking for changes, but if you are..I don't know this functionality well, but I think the config.get issue is not a problem for regular (not retried) submissions because if the method config/workflow no longer exists then the UI doesn't let you create submission because you need to go through a workflow?  If all that is correct, then I'd say let's do something similar in the UI and hide the re-try failure/aborted buttons if the workflow doesn't exist.  Would that work?